afkak/common.py

```diff
@@ -41,56 +42,383 @@
 ###############
 #   Structs   #
 ###############
+
+
+@attr.frozen(hash=False, eq=False)
+class BaseStruct:
+    """
+    Base class for all structs.
+    This class behaves like a tuple to maintain backwards compatibility
+    """
+
+    def __iter__(self):
+        return iter(attr.astuple(self, recurse=False))
+
+    def __getitem__(self, item):
+        return attr.astuple(self)[item]
+
+    def __len__(self):
+        return len(attr.astuple(self))
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return attr.astuple(self) == attr.astuple(other)
+        return NotImplemented
+
+    def __hash__(self):
+        return hash(attr.astuple(self))
+
+    def __lt__(self, other):
+        if isinstance(other, self.__class__):
+            return attr.astuple(self) < attr.astuple(other)
+        return NotImplemented
+
+
 # SendRequest is used to encapsulate messages and keys prior to
 # creating a message set
-SendRequest = namedtuple("SendRequest", ["topic", "key", "messages", "deferred"])
+@attr.frozen
+class SendRequest(BaseStruct):
+    """
+    Request to send a message to a topic
+
+    Attributes:
```

Comment:
We should have a consistent docstring format. Some of these new docstrings are using Google style docstrings and others are using the Sphinx style docstrings.